WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/NewDomainSearchViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package org.wordpress.android.ui.domains.management.newdomainsearch
+
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.wordpress.android.analytics.AnalyticsTracker
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.domains.management.newdomainsearch.domainsfetcher.NewDomainsSearchRepository
+import org.wordpress.android.ui.domains.management.newdomainsearch.domainsfetcher.ProposedDomain
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.viewmodel.ScopedViewModel
+import javax.inject.Inject
+import javax.inject.Named
+
+private const val SEARCH_QUERY_DELAY_MS = 250L
+
+@OptIn(FlowPreview::class)
+@HiltViewModel
+class NewDomainSearchViewModel @Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    private val newDomainsSearchRepository: NewDomainsSearchRepository,
+    analyticsTracker: AnalyticsTrackerWrapper
+) : ScopedViewModel(mainDispatcher) {
+    private val _actionEvents = MutableSharedFlow<ActionEvent>()
+    val actionEvents: Flow<ActionEvent> = _actionEvents
+
+    private val _uiStateFlow: MutableStateFlow<UiState> = MutableStateFlow(UiState.PopulatedDomains(emptyList()))
+    val uiStateFlow = _uiStateFlow.asStateFlow()
+
+    private val debouncedQuery = MutableStateFlow("")
```

Comment:
Yeah, I use the state flow because it doesn't notify the subscriber about duplicated values so we can get rid of useless API calls when the only difference in a query is whitespace.